### ISSUE TITLE
Feature request: Index maker

### ISSUE THREAD

**Issue by 5e-Cleric** [naturalcrit/homebrewery#1807]
So when i say index people thinks of the ToC, but i am talking about the actual index on the PHB:
![image](https://user-images.githubusercontent.com/91662974/140615723-7f74055c-d4de-4f80-867a-6ff809a84811.png)

Could something like this be implemented? something simmilar to the ToC, just getting a few keywords, maybe the ones with the class=index? or the ones we target with **** or -word-?

I feel it is the last bit to add for hb to really look like the PHB, though i get no more than a few people have brews that large as to need one of those.

**Comment by ericscheid** [comment]
Collecting a list of words and compiling a concordance isn't too hard, but also isn't very helpful.

A good index will have things like concept groups (`casting a spell` has sub-entries of `area of effect`, `attack roll`, etc), and cross references (like `carrying capacity: see under lifting and carrying`).

Indexing is difficult.

**Comment by calculuschild** [comment]
Yeah, I think making an autogenerator for an index is going to be very tricky to pull off.

However we can definitely add a snippet for an index with the necessary styling and some example text inside.

**Comment by 5e-Cleric** [comment]
oh man... my brew is 70+ pages long, it is a pain to track words, lucky we still have chrome's search for words

**Comment by ericscheid** [comment]
As a thought .. we might be able to leverage `{curly-spans}` here, at least as a means of marking up spans of text with a class.

> Every character has six ***{index*** ability scores}. These can be ***{index,topic="ability score",entry="rolling"*** rolled} or pulled from a ***{index,topic="ability score"*** standard array}, and these can be modified by ***{index,entry="ability score",entry="increase",see="racial traits"*** racial traits}.

**Comment by Gazook89** [comment]
@ericscheid I think you want to be sure you have double brackets on those spans e.g. `{{index ability scores}}`  otherwise you have got the injector syntax.

**Comment by 5e-Cleric** [comment]
This issue is closed, as will never be done, instead a template snippet will be done, as tracked in #1096

**Comment by ericscheid** [comment]
Re-opening as PR #3113 is making a brave attempt.

**Comment by dbolack-ab** [comment]
Working code needs an eval.
